compiler/compiler.go

```diff
@@ -1775,6 +1776,40 @@ func (b *builder) createExpr(expr ssa.Value) (llvm.Value, error) {
 			if expr.CommaOk {
 				valueType = valueType.(*types.Tuple).At(0).Type()
 			}
+
+			var index llvm.Value
+			var foundStringByteSliceConversion bool
+			if t, ok := expr.Index.Type().Underlying().(*types.Basic); ok && t.Info()&types.IsString != 0 {
+				if convert, ok := expr.Index.(*ssa.Convert); ok {
+					if slice, ok := convert.X.Type().Underlying().(*types.Slice); ok {
+						if elem, ok := slice.Elem().Underlying().(*types.Basic); ok {
+							if elem.Kind() == types.Byte {
+								foundStringByteSliceConversion = true
+
+								key := b.getValue(convert.X)
+
+								keyBuf := b.CreateExtractValue(key, 0, "map.keyBuf")
+								keyLen := b.CreateExtractValue(key, 1, "map.keyLen")
+
+								// Create the string
+								str := b.ctx.ConstStruct([]llvm.Value{
+									llvm.Undef(b.i8ptrType),
+									llvm.Undef(b.uintptrType),
+								}, false)
```

Comment:
This looks unsafe. A string is not just a struct, it's a `runtime._string` named struct. And I don't think these values you're inserting are actually constants.